homeassistant/components/lovelace/util.py

```diff
@@ -0,0 +1,25 @@
+"""Utils for Lovelace."""
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.util import slugify
+
+
+def url_slug(value: Any) -> str:
+    """Validate value is a valid url slug."""
+    if value is None:
+        raise vol.Invalid("Slug should not be None")
+    if "-" not in value:
+        raise vol.Invalid("Url path needs to contain a hyphen (-)")
+    str_value = str(value)
+    slg = slugify(str_value, separator="-")
+    if str_value == slg:
+        return str_value
+    raise vol.Invalid(f"invalid slug {value} (try {slg})")
+
+
+class ConfigNotFound(HomeAssistantError):
+    """When no config available."""
```

Comment:
I can see why we would move `url_slug` into a util, but moving the exception here, I don't follow.

Could you clarify why this was moved here?

../Frenck

Reply:
It just felt weird to have a class (even if it's an Exception class) defined in `const.py`
But I can move it back if that is prefered

Reply:
Or move it to a separate module (`common.py`)

Reply:
Exceptions often live in an exceptions module (more commonly across libraries).

Having it in the main module isn't weird IMHO 🤷 
Classifying it as a util, is what triggered me, as it isn't a util.